scan-payment/src/main/java/com/getbouncer/scan/payment/Image.kt

```diff
@@ -86,159 +94,182 @@ fun hasOpenGl31(context: Context): Boolean {
-            (this.height / 2) - heightOffset,
-            (this.width / 2) - widthOffset,
-            (this.height / 2) + heightOffset
-        ),
-        Rect(
-            (this.width / 2) - widthOffset,
-            (this.height / 2) - heightOffset,
-            (this.width / 2) + widthOffset,
-            (this.height / 2) + heightOffset
-        ),
-        Rect(
-            (this.width / 2) + widthOffset,
-            (this.height / 2) - heightOffset,
-            this.width,
-            (this.height / 2) + heightOffset
-        ),
-        Rect(
-            0,
-            (this.height / 2) + heightOffset,
-            (this.width / 2) - widthOffset,
-            this.height
-        ),
-        Rect(
-            (this.width / 2) - widthOffset,
-            (this.height / 2) + heightOffset,
-            (this.width / 2) + widthOffset,
-            this.height
-        ),
-        Rect(
-            (this.width / 2) + widthOffset,
-            (this.height / 2) + heightOffset,
-            this.width,
-            this.height
-        )
-    )
-    val toSegments = arrayOf(
-        Rect(
-            0,
-            0,
-            toBorderWidth,
-            toBorderWidth
-        ),
-        Rect(
-            toBorderWidth,
-            0,
-            toBorderWidth + toCenterDimension,
-            toBorderWidth
-        ),
-        Rect(
-            toBorderWidth + toCenterDimension,
-            0,
-            (2 * toBorderWidth + toCenterDimension),
-            toBorderWidth
-        ),
-        Rect(
-            0,
-            toBorderWidth,
-            toBorderWidth,
-            toBorderWidth + toCenterDimension
-        ),
-        Rect(
-            toBorderWidth,
-            toBorderWidth,
-            toBorderWidth + toCenterDimension,
-            toBorderWidth + toCenterDimension
-        ),
-        Rect(
-            toBorderWidth + toCenterDimension,
-            112,
-            (2 * toBorderWidth + toCenterDimension),
-            toBorderWidth + toCenterDimension
-        ),
-        Rect(
-            0,
-            toBorderWidth + toCenterDimension,
-            toBorderWidth,
-            (2 * toBorderWidth + toCenterDimension)
-        ),
-        Rect(
-            toBorderWidth,
-            toBorderWidth + toCenterDimension,
-            toBorderWidth + toCenterDimension,
-            (2 * toBorderWidth + toCenterDimension)
-        ),
-        Rect(
-            toBorderWidth + toCenterDimension,
-            toBorderWidth + toCenterDimension,
-            (2 * toBorderWidth + toCenterDimension),
-            (2 * toBorderWidth + toCenterDimension)
-        )
+    // Transforms the viewfinder from the preview to the image size
+    val projectedViewFinder = previewSize.projectRegionOfInterest(toSize = this.size(), regionOfInterest = cardFinder)
+    // Creates a square version of the viewfinder
+    val projectedViewFinderSquare = Size(projectedViewFinder.width(), projectedViewFinder.width())
```

Comment:
we don't have any guarantee that the `projectedViewFinderSquare` is within the bounds of the new image. Let's make sure it is.